.vib/nginx-ingress-controller/ginkgo/nginx-ingress-controller.go

```diff
@@ -0,0 +1,39 @@
+package integration
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	cv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	// For client auth plugins
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+var _ = Describe("NGINX Ingress Controller", func() {
+	var coreclient cv1.CoreV1Interface
+	var ctx context.Context
+
+	BeforeEach(func() {
+		coreclient = cv1.NewForConfigOrDie(clusterConfigOrDie())
+		ctx = context.Background()
+	})
+
+	Context("Testing ingress", func() {
+		var controllerPods v1.PodList
+		var containerLogs []string
+
+		BeforeEach(func() {
+			controllerPods = getPodsByLabelOrDie(ctx, coreclient, *namespace, "app.kubernetes.io/component=controller")
+			containerLogs = getContainerLogsOrDie(ctx, coreclient, *namespace, controllerPods.Items[0].GetName(), "controller")
+		})
+
+		It("is managed by nginx-ingress-controller", func() {
+			var toCheck = "Found valid IngressClass\" ingress=\"default/" + *ingressName
+			Expect(containsString(containerLogs, toCheck)).To(BeTrue())
```

Comment:
This effectively tests that NIC is able to recognize the Ingress resource, but maybe it could also be interesting to check that some of the rules defined in it work as expected (e.g. one can access one of the paths pointing to a backend service). This does not necessarily need to be performed in GinkGo, maybe GOSS/Cypress is easier. WDYT?

Reply:
I think what @joancafom proposed can be done easier with GOSS than with Cypress, not?

Reply:
After giving this some thought I ended up refactoring how the whole asset is tested, now using a mock-up deployment/service/ingress to simulate a real use case. As such, this comment is now obsolete. Please review the new tests when possible @joancafom @agomezmoron